app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -76,7 +79,12 @@ public void updateResourceManagerAddresses() {
         updateAuthToken();
         for (String id : ids) {
           try {
-            String resourceManager = configuration.get(RESOURCE_MANAGER_ADDRESS + "." + id);
+            String resourceManager;
+            if (resourceManagerHttpPolicy.equals("HTTPS_ONLY")){
+              resourceManager = "https://" + configuration.get(RESOURCE_MANAGER_HTTPS_ADDRESS + "." + id);
+            } else {
+              resourceManager = "http://" + configuration.get(RESOURCE_MANAGER_HTTP_ADDRESS + "." + id);
```

Comment:
Set a default value while retrieving configuration...here and elsewhere

probably localhost:8088